superset/models/helpers.py

```diff
@@ -1350,6 +1377,9 @@ def values_for_column(
         if self.fetch_values_predicate:
             qry = qry.where(self.get_fetch_values_predicate(template_processor=tp))
 
+        if rls_filters := self.get_rls_filters_for_column(column_name, tp):
+            qry = qry.where(or_(*rls_filters))
```

Comment:
are we sure the content of the `OR` clause will be wrapped in parenthesis?

Reply:
I'm pretty sure SQLAlchemy does the right thing here, but a unit test would be good. Preferably one where one of the filters is `cond1 AND cond2`, ensuring the generated SQL is:

```sql
(cond1 AND cond2) OR cond3
```

And not:

```sql
cond1 AND cond2 OR cond3
```